cmd/cloudshell_open/cloudrun.go

```diff
@@ -115,9 +115,9 @@ func envVars(project, name, region string) (map[string]struct{}, error) {
 // prevent deployment failures due to Cloud Run service naming constraints such
 // as:
 //
-//   * names with a leading non-letter (e.g. digit or '-') are prefixed
-//   * names over 63 characters are truncated
-//   * names ending with a '-' have the suffix trimmed
+//   - names with a leading non-letter (e.g. digit or '-') are prefixed
+//   - names over 63 characters are truncated
+//   - names ending with a '-' have the suffix trimmed
 func tryFixServiceName(name string) string {
 	if name == "" {
```

Comment:
Being very nitpicky here, but this seems like this would be an erroneous situation, and a function name that starts with `try` sort of implies to me that it should return errors for unexpected things (maybe just baggage from working with too many languages with try/catch), but otherwise, why even guard here? Either we never expect this to happen or else it's probably better to fail here than when creating the image fails with an empty service name.

Maybe better if we have this?

```go
func tryFixServiceName(name string) (string, error) {
        if name == "" {
                return "", fmt.Errorf("service name can't be empty")
        }
```

Reply:
There's a lot of older code here that I don't think would pass Golang standards today. The above diff is specifically the output of a linting command, which happens to be near one of the many other issues. I won't be addressing all of those in this PR

Reply:
Sorry, conversation isn't resolved yet. I think you thought I was commenting on the lint recommendations. I was commenting on and suggesting that `tryServiceName` actually return an error rather than accept and return an (erroneous) empty string. I realize you didn't touch this code, my suggestion is just based on us leaving code in a better state than we found it.

Reply:
Hi @glasnt, the rest of the PR LGTM and I don't want to be too nitpicky. I asked @telpirion to take a look at this and resolve the conversation accordingly.